packages/xpetra/src/MultiVector/Xpetra_MultiVectorFactory_def.hpp

```diff
@@ -64,8 +65,8 @@ Build(const Teuchos::RCP<const Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node>>&
 {
     XPETRA_MONITOR("MultiVectorFactory::Build");
 
-    RCP<const BlockedMap<LocalOrdinal, GlobalOrdinal, Node>> bmap =
-        Teuchos::rcp_dynamic_cast<const BlockedMap<LocalOrdinal, GlobalOrdinal, Node>>(map);
+    RCP<const Xpetra::BlockedMap<LocalOrdinal, GlobalOrdinal, Node>> bmap =
```

Comment:
This looks like the only place that actually needs the definition of `Xpetra::BlockedMap` (since the target type [must be a complete type](https://en.cppreference.com/w/cpp/language/dynamic_cast)).

Reply:
Do people see the trouble that all these template parameters are causing us?  Some piece of code that should be trivial is costing hours and hours of multiple developers' time.  None of these three template parameters have ever added value for downstream solvers or applications.

@trilinos/framework @prwolfe @jwillenbring @bartlettroscoe @micahahoward @rrdrake I want people to see this.

Reply:
@trilinos/tpetra @kddevin